src/Features/CSharp/Portable/SignatureHelp/InvocationExpressionSignatureHelpProvider.cs

```diff
@@ -72,76 +74,77 @@ private static bool IsArgumentListToken(InvocationExpressionSyntax expression, S
-
-            var structuralTypeDisplayService = document.Project.LanguageServices.GetRequiredService<IStructuralTypeDisplayService>();
-            var documentationCommentFormattingService = document.Project.LanguageServices.GetRequiredService<IDocumentationCommentFormattingService>();
-
-            var textSpan = SignatureHelpUtilities.GetSignatureHelpSpan(invocationExpression.ArgumentList);
-            var syntaxFacts = document.GetRequiredLanguageService<ISyntaxFactsService>();
-
-            if (methodGroup.Any())
+            else
             {
-                var accessibleMethods = GetAccessibleMethods(invocationExpression, semanticModel, within, methodGroup, cancellationToken);
-                var (items, selectedItem) = await GetMethodGroupItemsAndSelectionAsync(accessibleMethods, document, invocationExpression, semanticModel, symbolInfo, cancellationToken).ConfigureAwait(false);
-
-                return CreateSignatureHelpItems(
-                    items,
-                    textSpan,
-                    GetCurrentArgumentState(root, position, syntaxFacts, textSpan, cancellationToken),
-                    selectedItem);
+                // get the candidate methods
+                var symbolDisplayService = document.GetLanguageService<ISymbolDisplayService>();
+                methods = semanticModel.GetMemberGroup(invocationExpression.Expression, cancellationToken)
+                                               .OfType<IMethodSymbol>()
+                                               .ToImmutableArray()
+                                               .FilterToVisibleAndBrowsableSymbols(options.HideAdvancedMembers, semanticModel.Compilation);
+                methods = GetAccessibleMethods(invocationExpression, semanticModel, within, methods, cancellationToken);
+                methods = methods.Sort(semanticModel, invocationExpression.SpanStart);
+
+                if (!methods.Any())
+                {
+                    return null;
+                }
+
+                currentSymbol = semanticModel.GetSymbolInfo(invocationExpression, cancellationToken).Symbol as IMethodSymbol;
             }
 
-            var invokedType = semanticModel.GetTypeInfo(invocationExpression.Expression, cancellationToken).Type;
-            if (invokedType is INamedTypeSymbol expressionType && expressionType.TypeKind == TypeKind.Delegate)
-            {
-                var items = GetDelegateInvokeItems(invocationExpression, semanticModel, structuralTypeDisplayService,
-                    documentationCommentFormattingService, within, expressionType, out var selectedItem, cancellationToken);
+            // guess the best candidate if needed and determine parameter index
+            var arguments = invocationExpression.ArgumentList.Arguments;
+            LightweightOverloadResolution.RefineOverloadAndPickParameter(document, position, semanticModel, methods, arguments, ref currentSymbol, out var parameterIndex);
```

Comment:
I will try to re-organize this code to split the delegate/function-pointer scenarios to another method